source/client/RunManagement.py

```diff
@@ -0,0 +1,250 @@
+                aces_list[0].tempnumber = -1
+                card_group.insert(0, aces_list[0])
+                aces_list = []
+            elif card_group[-1].tempnumber == 12 and not isWild(card_group[-1], wild_numbers) \
+                    and not isWild(card_group[-2], wild_numbers):
+                this_wild = groups_wilds.pop(-1)
+                this_wild.tempnumber = 13
+                card_group.append(this_wild)
+                aces_list[0].tempnumber = 14
+                card_group.append(aces_list[0])
+                aces_list = []
+    if len(aces_list) > 0:
+        raise Exception('Ace cannot be played')
+   # Calculate if wild cards can be played  - only possible remaining slots for wilds are at the ends of the current run.
+    num_remaining_wilds = len(groups_wilds)
+    possible_wild_assignments = []
+    if not num_remaining_wilds == 0:
+        # are there options on where to assign wilds?
+        if num_remaining_wilds > 2:
+            raise Exception('You cannot play all the wild cards.')
+        if card_group[0].tempnumber > 1 and not isWild(card_group[0], wild_numbers) \
+                and not isWild(card_group[1], wild_numbers):
+            runslot = card_group[0].tempnumber - 1
+            possible_wild_assignments.append(runslot)
+        if card_group[-1].tempnumber < 14 and not isWild(card_group[-1], wild_numbers) \
+                and not isWild(card_group[-2], wild_numbers):
+            runslot = card_group[-1].tempnumber + 1
+            possible_wild_assignments.append(runslot)
+        # Can wilds be played?
+        if num_remaining_wilds > len(possible_wild_assignments):
+            raise Exception('you cannot play all the wild cards.')
+        num_remaining_wilds = len(groups_wilds)
+        # Can wilds be automatically played?
+        if num_remaining_wilds == 2:
+            groups_wilds[0].tempnumber = possible_wild_assignments[0]
+            groups_wilds[1].tempnumber = possible_wild_assignments[1]
+            card_group.insert(0, groups_wilds[0])
+            card_group.append(groups_wilds[1])
+            groups_wilds=[]
+            possible_wild_assignments=[]
+        elif num_remaining_wilds == 1 and len(possible_wild_assignments) == 1:
+            groups_wilds[0].tempnumber = possible_wild_assignments[0]
+            if possible_wild_assignments[0] < card_group[0].tempnumber:
+                card_group.insert(0, groups_wilds[0])
+            else:
+                card_group.append(groups_wilds[0])
+            groups_wilds=[]
+            possible_wild_assignments=[]
+        num_remaining_wilds = len(groups_wilds)
+        if num_remaining_wilds == 0:
+            possible_wild_assignments = []
+        # if num_remaining_wilds == 1 and len(possible_wild_assignments) == 2:
+        # Will need to ask player whether to play high or low, controller handles that.
+    # Final rules check in processRuns: double check that wilds are not placed too close together.
```

Comment:
If this is a helper it can be run both before you mess around with aces, which might catch some bad runs faster, and at the end.

Reply:
Rearranged code.  Hard to catch this earlier, as number of wilds remaining at end depends on where they're needed.